src/encode/pattern/mod.rs

```diff
@@ -650,6 +784,34 @@ impl PatternEncoder {
         PatternEncoder {
             chunks: Parser::new(pattern).map(From::from).collect(),
             pattern: pattern.to_owned(),
+            color_map: ColorMap::default(),
+        }
+    }
+
+    /// Creates a `PatternEncoder` from a pattern string and color hashmap.
+    ///
+    /// The pattern string syntax is documented in the `pattern` module.
+    ///
+    ///
+    ///! ```no_run
+    ///! # #[cfg(all(feature = "console_appender",
+    ///! #           feature = "file_appender",
+    ///! #           feature = "pattern_encoder"))]
+    ///! # fn f() {
+    ///! // Change color of Info level msgs, other msgs will be the default color
+    ///! let mut log_color_map = ColorMap::default()
+    ///! log_color_map.info_color = Some(log4rs::encode::Color::Cyan);
+    ///! let pattern_encoder = PatternEncoder::new_with_colormap("{d} - {m}{n}", log_color_map);
+    ///!
+    ///! }
+    ///! # }
+    ///! # fn main() {}
+    ///! ```
+    pub fn new_with_colormap(pattern: &str, color_map: ColorMap) -> PatternEncoder {
```

Comment:
Feels weird to combine new and colors here.
How about keep new, and add a `with_color` method ? 
Something along this: 
`PatternEncoder::new(pattern).with_color(color_map)`